src/compas_timber/elements/slab.py

```diff
@@ -1,284 +1,215 @@
-from compas.data import Data
-from compas.geometry import Box
-from compas.geometry import Brep
-from compas.geometry import Frame
-from compas.geometry import Line
-from compas.geometry import Plane
+from compas.geometry import Point
 from compas.geometry import Polyline
-from compas.geometry import bounding_box
+from compas.geometry import Transformation
 from compas_model.elements import Element
 
-from compas_timber.utils import classify_polyline_segments
+from .plate_geometry import PlateGeometry
 
 
-class OpeningType(object):
-    DOOR = "door"
-    WINDOW = "window"
+class SlabType(object):
+    """Constants for different types of slabs.
 
+    Attributes
+    ----------
+    WALL : str
+        Constant for wall slabs.
+    FLOOR : str
+        Constant for floor slabs.
+    ROOF : str
+        Constant for roof slabs.
+    GENERIC : str
+        Constant for generic slabs.
+    """
 
-class Opening(Data):
-    @property
-    def __data__(self):
-        return {
-            "polyline": self.polyline,
-            "opening_type": self.opening_type,
-        }
-
-    def __init__(self, polyline, opening_type, **kwargs):
-        super(Opening, self).__init__(**kwargs)
-        self.polyline = polyline
-        self.opening_type = opening_type
-
-    def __repr__(self):
-        return "Opening(type={})".format(self.opening_type)
-
-    def orient_polyline(self, normal):
-        self.polyline = _oriented_polyline(self.polyline, normal)
-
-
-class Slab(Element):
-    """Represents a single timber wall element.
-    Serves as container for beams joints and other related elements and groups them together to form a wall.
-
-    Wall is often a single unit of prefabricated timber wall element.
-    It is often refered to as an enveloping body.
-
-    TODO: complete this docstring
+    WALL = "wall"
+    FLOOR = "floor"
+    ROOF = "roof"
+    GENERIC = "generic"
+
+
+class Slab(PlateGeometry, Element):
+    """Represents a timber slab element (wall, floor, roof, etc.).
+
+    Serves as container for beams, joints and other related elements and groups them together to form a slab.
```

Comment:
technically, the slab knows nothing about the joints. these are still only described at the model level. the only sub-elements of a slab are other `TimberElement` or `Element` objects.